generators/app/index.js

```diff
@@ -189,8 +194,44 @@ module.exports = class extends BaseGenerator {
 
         this.withEntities = this.options['with-entities'];
         this.skipChecks = this.options['skip-checks'];
-        const blueprint = jhipsterUtils.normalizeBlueprintName(this.options.blueprint || this.config.get('blueprint'));
-        this.blueprint = this.configOptions.blueprint = blueprint;
+
+        // check for old single blueprint declaration
+        let blueprint = this.options.blueprint;
```

Comment:
I dont think we need to do many checks. this can be simplified to
```suggestion
        let blueprints = this.options.blueprints || '';
        let blueprint = this.options.blueprint;
        if (blueprint) {
            this.warning('--blueprint option is deprecated. Please use --blueprints instead');
            if (!blueprints.split(',').includes(blueprint)) {
                blueprints = `${blueprint},${blueprints}
            }
        }
        if (!blueprints) {
            // load blueprints from config file
            const blueprintName = this.config.get('blueprint');
            if (blueprintName) {
                const blueprintVer = this.config.get('blueprintVersion') || 'latest';
                blueprint = {
                    name: blueprintName,
                    version: blueprintVer
                };
            }

            blueprints = this.config.get('blueprints') || [];
            if (blueprint) {
                blueprints.unshift(blueprint);
            }
        }
```
```

Reply:
Fixed